assets/src/edit-story/components/keyboardShortcutsMenu/karma/keyboardShortcuts.karma.js

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { within } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+
+describe('Keyboard Shortcuts Menu', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+    fixture.setFlags({ showKeyboardShortcutsButton: true });
+
+    await fixture.render();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  describe('CUJ: User can interact with menu using mouse: Click toggle button to open, click close button to close menu', () => {
+    it('should be able to open menu by clicking on keyboard shortcut button', async () => {
+      const hiddenMenu = await fixture.screen.queryByTestId(
+        'keyboard-shortcuts-menu'
+      );
+      // Menu should be closed
+      expect(hiddenMenu).toBeNull();
+
+      const menuToggle = fixture.screen.getByTestId(
+        'keyboard-shortcuts-toggle'
+      );
```

Comment:
I'd prefer for this (and all other element access) to go through `fixture.editor` and the whole fake DOM structure we've created there - it makes for cleaner and more consistent tests.

This button would be a child of the carousel, but not sure where the modal best fits in.

Secondly, please use `getByRole` whenever possible - including this case. `data-testid` is really only for cases, where role-based access cannot work (which will be very rare).

Reply:
Coolio!  I removed the `data-testId` calls and tags in the code and replaced all the lookups with `ByRole`.  I also switched all the editor buttons to use `fixture.editor`, but since the modal itself is completely outside the editor's container, I had to stick with `fixture.screen` to access the modal and its contents.